spec/ios_update_metadata_source_spec.rb

```diff
@@ -0,0 +1,13 @@
+require 'spec_helper'
+require 'shared_examples_for_update_metadata_source_action'
+
+describe Fastlane::Actions::IosUpdateMetadataSourceAction do
+  before do
+    # This works around the `ensure_git_status_clean` call within the action.
+    # It would be quite cumbersome to iterate on the code if we had to commit
+    # every change _before_ running the tests...
+    allow(Fastlane::Actions::EnsureGitStatusCleanAction).to receive(:run)
```

Comment:
Gosh... I started looking into this but never wrote about it and now I'm not sure where it was 🤔 

I run into a circular issue. Our tests already use `tmp` directories, but because they also create files in there, the status of that Git repository appears dirty. The solution would be to read the inputs and output the generated `.po` in a location that doesn't have a Git repo, while also running the tests in a different folder that has a git repo, but clean. (_Something along those lines, I'm writing this from memory without searching for my notes or commits_)

Not sure it's worth for the sake of this one test.

If this smell is too smelly to merge, I think we can live with the codebase not having this test. After all, if we'll ever need to modify the existing code, I'd very much hope we'd take the time to implement the rewrites we discussed, which would make this test redundant.

Reply:
> The solution would be to read the inputs and output the generated .po in a location that doesn't have a Git repo, while also running the tests in a different folder that has a git repo, but clean. (Something along those lines, I'm writing this from memory without searching for my notes or commits)

I feel like this would be even dirtier. Like, it would be trying to trick the system by running the action in a different folder than the thing it's working on… which tbh if that dirty trick worked, could actually be considered a bug, because we'd probably instead want to check that the repo root _corresponding to the input folder_ would not be dirty — as opposed to the pwd — since the whole point I think was to ensure we did not forget to commit some of the input files the generated `.po` was based on… So I'd vote against such a hack.

I think we should:
1. Either keep the mocking and update the comment to explain why it's necessary (i.e. because in the special context of tests we create temp files just for the sake of the test, and in a temp dir that is not tracked in git anyway)
2. Or make the temp test dir be a git repo (after all we do something similar in our `git_helper_spec.rb`… though that spec is specifically about git so that make sense there but less sense here…)
3. Or remove the test from the codebase for now if we feel it's too smelly to merge.

Personally I'm ok with (1).